lib/vdsm/virt/cpumanagement.py

```diff
@@ -136,6 +138,27 @@ def _assign_shared(cif, target_vm=None):
                 # pool for other VMs.
 
 
+def _flatten_cpusets(cpusets_dict):
+    """
+    Convert dictionary with cpusets (key: vCPU ID, value: frozenset) into a
+    list where on index <vCPU_ID> is string description of cpuset if such is
+    specified in the dictionary and None otherwise. The first form is used
+    when parsing libvirt domain XML and the second form is used in VDSM API
+    calls.
+
+    :param cpusets_dict: Dictionary with CPU sets
+    :type cpusets_dict: dict
+
+    :returns: list of strings
+    """
+    if len(cpusets_dict.keys()) == 0:
+        return []
+    result = [None] * (max(cpusets_dict.keys()) + 1)
+    for id, cpuset in cpusets_dict.items():
```

Comment:
Better to use `id_` or `cpu_id` to distinguish it from `id` built-in.

Reply:
done

Reply:
Needs an update also on the following line.

Reply:
right, what was I thinking...